src/code42cli/cmds/devices.py

```diff
@@ -330,13 +342,61 @@ def list_devices(
         df = _add_settings_to_dataframe(state.sdk, df)
     if include_usernames:
         df = _add_usernames_to_device_dataframe(state.sdk, df)
+    if include_legal_hold_membership:
+        df = _add_legal_hold_membership_to_device_dataframe(state.sdk, df)
     if df.empty:
         click.echo("No results found.")
     else:
         formatter = DataFrameOutputFormatter(format)
         formatter.echo_formatted_dataframe(df)
 
 
+def _add_legal_hold_membership_to_device_dataframe(sdk, df):
+    matters = _get_all_active_matters(sdk)
+    matters = [matter["legalHoldUid"] for matter in matters]
+
+    legal_hold_members = []
+
+    for matter in matters:
+        memberships = _get_legal_hold_memberships_for_matter(
+            sdk, matter_id=matter, active=None
+        )
+
+        legal_hold_members.extend(
+            [
+                [
+                    str(member["active"]),
+                    member["user"]["userUid"],
+                    member["legalHold"]["legalHoldUid"],
+                    member["legalHold"]["name"],
+                ]
+                for member in memberships
+            ]
+        )
+
+    legal_hold_member_dataframe = (
+        DataFrame.from_records(
+            legal_hold_members,
+            columns=[
+                "legalHoldMemberActive",
+                "userUid",
+                "legalHoldUid",
+                "legalHoldName",
+            ],
+        )
+        .groupby(["userUid"])
+        .agg(lambda col: ",".join(col))
+    )
+    df = df.merge(legal_hold_member_dataframe, how="left", on="userUid")
+
+    df.loc[
+        df["status"] == "Deactivated",
+        ["legalHoldMemberActive", "legalHoldUid", "legalHoldName"],
+    ] = np.nan
```

Comment:
Instead of using np.nan for null values here, let's use an empty string, I think `NaN` would confuse people in the output and think maybe there was a bug. To replace any np.nan values resulting from the `.groupby()`, you can call `.fillna(value="")` on the df to replace those as well.

Reply:
I chose `NaN` to match the value assumed by pandas of an empty value. `NaN`'s are generated when the legal hold table is merged with the devices table. If I use the above solution prior to merging with the devices table, the legalHoldUid and legalHold.name columns will contain a mix of None's, actual values, and NaNs.

To confirm: which of the following would you prefer?

* NaN's in table view
* None's in table view

(Either will produce null values in the other output formats)

Reply:
Oops, I meant to say "resulting from the`.merge()`" instead of `.groupby()`. Yeah, the `.fillna(value="")` needs to be called after the merge. 

I think we should convert NaN/None values to an empty string for table/CSV outputs, but we'd probably want to keep them as NaN/None when we output to json so it translates to `null`. But we should push that logic to the `DataFrameOutputFormatter` so we only need to handle it in one place for any commands that use DataFrames.

I'll start a new PR to rework the formatter to handle that, since I'd like to refactor it a bit for clarity and add some tests around handling dataframe nulls.

Reply:
NaNs removed

Reply:
Ok, I've opened https://github.com/code42/code42cli/pull/245 to add the null handling in the DataFrameOutputFormatter, so we don't need to be concerned with converting them here, since we want the conversion to be conditional on the output format chosen by the end-user. 

Sorry to have you flip/flop this!

Reply:
Resolved, and awaiting #245